### ISSUE TITLE
Kotlin/Native Plugin for IntelliJ confuses method signatures when implementing ObjC protocol

### ISSUE THREAD

**Issue by chris-hatton** [JetBrains/kotlin-native#2476]
**Affects**
IntelliJ Community 2018.3
`kotlin-multiplatform` plugin versions`1.3.10` & `1.3.11`
```
class Source : NSObject(), UITableViewDataSourceProtocol {
    override fun tableView(tableView: UITableView, cellForRowAtIndexPath: NSIndexPath): UITableViewCell {...}
    override fun tableView(tableView: UITableView, numberOfRowsInSection: NSInteger): NSInteger {...}
}
```
<img width="1092" alt="screenshot 2018-12-18 at 10 51 15" src="https://user-images.githubusercontent.com/895683/50150744-c55caf80-02b6-11e9-9406-139d7340db09.png">
This is a valid delcaration which **compiles without error**.

However the IDE plugin appears to think a different method signature is being overridden and complains that the return type is incorrect.

**Comment by Alex009** [comment]
on kotlin 1.3.50 same issue :(

**Comment by trondat** [comment]
This is not resolved in Kotlin 1.3.61 either.. :-(..

**Comment by SvyatoslavScherbina** [comment]
Moved to https://youtrack.jetbrains.com/issue/KT-48067, closing.
